### PR TITLE
Add unit test for StoreToJobLister

### PR DESCRIPTION
**Pull request by mqliang**
This PR add unit test for StoreToJobLister.

### CODE REVIEW

File: pkg/client/cache/listers_test.go [hunk 1]
```diff
@@ -278,6 +278,136 @@ func TestStoreToDaemonSetLister(t *testing.T) {
...
+		},
+	}
+	for _, c := range testCases {
+		for _, r := range c.inJobs {
+			store.Add(r)
+		}
+
+		Jobs, err := c.list()
+		if err != nil && c.expectErr {
+			continue
```
Comment (krousey) [inline pkg/client/cache/listers_test.go]: Shouldn't there be an error case here too?

Comment (mqliang) [inline pkg/client/cache/listers_test.go]: What do you mean  "an err case"?

Comment (mqliang) [inline pkg/client/cache/listers_test.go]: In such a case, we expect c.list() to return an error, so just continue.

Comment (krousey) [inline pkg/client/cache/listers_test.go]: Sorry I was reading this wrong. Thanks.

**Comment by mqliang** [comment]
@krousey Some other test function in this file use `Fatal()`, instead of `Error()` and `continue`. I have helped fix them in this PR https://github.com/kubernetes/kubernetes/pull/15815. Great thanks for your review!

**Comment by wojtek-t** [comment]
@mqliang - it seems there is some problem with your PR - Shippable is reporting:

    _output/local/go/src/k8s.io/kubernetes/pkg/client/cache/listers_test.go:297: cannot use extensions.JobList as type []extensions.Job in return argument
    _output/local/go/src/k8s.io/kubernetes/pkg/client/cache/listers_test.go:310: cannot use extensions.JobList as type []extensions.Job in return argument

Can you please fix that?

**Comment by wojtek-t** [comment]
@mqliang - thx!

**Pull request merged**